lib/commands/context.js

```diff
@@ -73,7 +80,121 @@ extensions.getNewRemoteDebugger = async function getNewRemoteDebugger () {
     garbageCollectOnExecute: util.hasValue(this.opts.safariGarbageCollect)
       ? !!this.opts.safariGarbageCollect
       : false,
-  });
+    udid: this.opts.udid,
+  }, this.isRealDevice());
+};
+
+commands.setContext = async function setContext (name, callback, skipReadyCheck) {
+  function alreadyInContext (desired, current) {
+    return (desired === current ||
+           (desired === null && current === NATIVE_WIN) ||
+           (desired === NATIVE_WIN && current === null));
+  }
+
+  log.debug(`Attempting to set context to '${name}'`);
+  if (alreadyInContext(name, this.curContext)) {
+    // already in the named context, no need to do anything
+    return;
+  }
+  if (name === NATIVE_WIN || name === null) {
```

Comment:
should we use util.hasValue instead of check null equality?

Reply:
or !._isString(name)

Reply:
No, this is supposed to handle the case where the context is set to `null`, which I'm not sure we've publicized but is supposed to do the same as setting it to `'NATIVE_APP'`. We've never had the same functionality for `undefined`, and I don't want to start.

Reply:
it would be probably much more readable to have isNativeContext method, which compares the given variable to null/NATIVE_WIN